docs/configuration-settings.md

```diff
@@ -0,0 +1,59 @@
+|  | CacheSlidingExpiration | E.g., `"0:15:00"` | Cache entry will expire if it hasn't been accessed in a set amount of time. Used, if CacheAbsoluteExpiration was not defined.
+|  | CacheAbsoluteExpiration | E.g., `"0:5:00"` | Cache entry will expire after a set amount of time. Used, if _RedisConnectionString_ not specified.
+| Notifications | | | Notifications configuration for `VirtoCommerce.Notifications` module
+|  | Gateway | E.g., `"SendGrid"` | Current notification sending gateway. Out of the box implemented and supported values: Smtp, SendGrid.
+|  | DefaultSender | E.g., `"noreply@gmail.com"` | **Required.** Sender identification used by current notification sending gateway.
+|  | Smtp | E.g.,<br>"Smtp": {<br>"SmtpServer": "http://smtp.gmail.com",<br>"Port": 587,<br>"Login": "my-login",<br>"Password": "my-password"<br>} | Smtp gateway configuration. Used, if `"Gateway": "Smtp"`
+|  | SendGrid | E.g.,<br>"SendGrid": {<br>"ApiKey": "my-SendGrid-Api-Key"<br>} | SendGrid gateway configuration. Used, if `"Gateway": "SendGrid"`
+| Search | | | Full text search configuration for `VirtoCommerce.Search` module
+|  | Provider | E.g., `"Lucene"` | **Required.** Current Search Provider. Implemented (in dedicated modules) and supported values: Lucene, AzureSearch, ElasticSearch.
+|  | Scope | E.g., `"default"`   | **Required.** Scope to use
+|  | Lucene | E.g.,<br>"Lucene": {<br>"Path": "App_Data/Lucene"<br>} | Lucene Provider configuration for _VirtoCommerce.LuceneSearch_ module. Used, if `"Provider": "Lucene"`
+|  | AzureSearch | E.g.,<br>"AzureSearch": {<br>"SearchServiceName": "my-ServiceName",<br>"Key": "my-AccessKey"<br>} | AzureSearch Provider configuration for _VirtoCommerce.AzureSearch_ module. Used, if `"Provider": "AzureSearch"`
+|  | ElasticSearch | E.g.,<br>"ElasticSearch": {<br>"Server": "localhost:9200",<br>"User": "elastic",<br>"Key": "",<br>"EnableHttpCompression": ""<br>} | ElasticSearch  Provider configuration for _VirtoCommerce.ElasticSearch_ module. Used, if `"Provider": "ElasticSearch"`
+| Content | | | **Required.** Static content (incl. themes) configuration for `VirtoCommerce.Content` module
+|  | Provider | `"FileSystem"` | Current Content (files) Provider. Supported values: FileSystem, AzureBlobStorage.
+|  | FileSystem | E.g., <br> "FileSystem": {<br> "RootPath": "~/cms-content",<br>"PublicUrl": "http://localhost:10645/cms-content/"<br>}  | File system based content provider configuration. This is the default provider, used if `AzureBlobStorage` is not set as current provider.
+|  | AzureBlobStorage | E.g., <br> "AzureBlobStorage": {<br> "ConnectionString": "",<br>"CdnUrl": ""<br>}  | Azure Blob Storage based content provider configuration. Used, if `"Provider": "AzureBlobStorage"`.
+
+
+**_Note:_** Each setting is optional, if it's not marked as **Required.**
+
+## Hierarchical keys
+* Within the Configuration API, a colon separator (`:`) works on all platforms.
```

Comment:
@eMazeika  Can you add common snippets/screenshot for: 
- Local Machine (Config)
- Environment Variable (Docker)
- Azure